monetdbe/_cffi/internal.py

```diff
@@ -256,9 +276,21 @@ def append(self, table: str, data: Mapping[str, np.ndarray], schema: str = 'sys'
 
     def prepare(self, query: str) -> monetdbe_statement:
         self._switch()
+
         stmt = ffi.new("monetdbe_statement **")
-        check_error(lib.monetdbe_prepare(self._monetdbe_database, str(query).encode(), stmt, ffi.NULL))
-        return stmt[0]
+        p_result = ffi.new("monetdbe_result **")
+        check_error(lib.monetdbe_prepare(self._monetdbe_database, str(query).encode(), stmt, p_result))
+
+        input_parameter_info = list()
+
+        columns = list(map(lambda x: result_fetch(p_result[0], x), (0, 2, 3, 6)))  # type: ignore[union-attr]
+
+        for r in range(p_result[0].nrows):
+            if (extract(columns[2], r)) is None:
+                row = (extract(columns[0], r), extract(columns[1], r), extract(columns[3], r))
```

Comment:
isn't it possible to do this in one for loop? I'm getting a bit confused by the column numbering magic also.

maybe we should introduce a named tuple for this like

https://github.com/MonetDBSolutions/MonetDBe-Python/blob/master/monetdbe/connection.py#L17

Reply:
Yes, Pushed!